src/main/scala/no/ndla/searchapi/service/search/SearchConverterService.scala

```diff
@@ -637,11 +637,25 @@ trait SearchConverterService {
     private def getFilters(taxonomyElement: TaxonomyElement,
                            subject: TaxSubject,
                            bundle: TaxonomyBundle,
-                           objectFilterConnections: List[FilterConnection]): List[SearchableTaxonomyFilter] = {
+                           objectFilterConnections: List[FilterConnection],
+                           filterVisibles: Boolean): List[SearchableTaxonomyFilter] = {
       val subjectFilters = bundle.filters.filter(_.subjectId == subject.id)
+      val visibleFilters = if (filterVisibles) {
+        subjectFilters.filter(f => {
+          if (f.metadata.isDefined) f.metadata.exists(_.visible)
+          else true
+        })
```

Comment:
Syns kanskje dette er enklere å lese?
```suggestion
        subjectFilters.filter(_.metadata.map(_.visible).getOrElse(true))
```